theano/tensor/nnet/tests/test_corr.py

```diff
@@ -259,6 +260,40 @@ def test_wrong_input(self):
         self.assertRaises(Exception, self.validate, (3, 2, 8, 8), (4, 2, 5, 5),
                           'valid', input=T.dtensor3())
 
+    def dtype_upcast(self):
+        """
+        Checks dtype upcast for CorrMM methods.
+        """
+        def rand(shape, dtype='float64'):
+            r = numpy.asarray(numpy.random.rand(*shape), dtype=dtype)
+            return r * 2 - 1
+
+        ops = [corr.CorrMM, corr.CorrMM_gradWeights, corr.CorrMM_gradInputs]
+        a_shapes = [[4, 5, 6, 3], [1, 5, 6, 3], [1, 5, 6, 3]]
+        b_shapes = [[7, 5, 3, 2], [1, 5, 3, 1], [7, 1, 3, 1]]
+        dtypes = ['float32', 'float64']
+
+        for op, a_shape, b_shape in zip(ops, a_shapes, b_shapes):
+            for a_dtype in dtypes:
+                c_dtype = 'float32'
+                for b_dtype in dtypes:
+                    if a_dtype == 'float32':
+                        a_tens = T.ftensor4()
+                    else:
+                        c_dtype = 'float64'
+                        a_tens = T.dtensor4()
+                    if b_dtype == 'float32':
+                        b_tens = T.ftensor4()
+                    else:
+                        c_dtype = 'float64'
+                        b_tens = T.dtensor4()
```

Comment:
you can simplify the code by doing:

    b_tens = T.tensor4(dtype=b_dtype)

Idem at other places.